### PR TITLE
fix encode by adding decode first

### LINKED ISSUE #34: Already encoded URLs are encoded twice
A bug introduced by #28 makes already encoded URLs encoded twice:

`https://en.wikipedia.org/wiki/Glob_%28programming%29` becomes `https://en.wikipedia.org/wiki/Glob_%2528programming%2529`

To prevent this side effect, it's required to check if the URL is already encoded.
The simple way is to decode first. If the url is not encoded, the decoded will be the same, if it's encoded it will provide a clean version of the string.
The computation required to decode is very small and not even noticeable.

This was first reported as a markdown-link-check issue tcort/markdown-link-check#142 and it will be required to update the dependency once fixed.

### PR DESCRIPTION
**Pull request by NicolasMassart**
Decode URL before encoding it to make sure we are not encoding twice.

- If the URL is already encoded, decode will revert the encoding and then the second encoding will make it look ok.
- If the URL is not encoded, decode will do nothing and encoding will work as usual.

The decode step is fast enough to be ignored for performances in our use case.

fixes #34

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,8 +1,1849 @@
 {
   "name": "link-check",
   "version": "4.5.3",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (nschonni) [inline package-lock.json:4]: Unrelated change from installing with NPM 7

File: lib/proto/http.js [hunk 2]
```diff
@@ -27,7 +27,8 @@ module.exports = {
         let fallbackRetryDelayInMs = ms(opts.fallbackRetryDelay || '60s');
 
         const options = {
-            uri: encodeURI(link),
+            // Decoding and encoding is required to prevent encoding already encoded URLs
+            uri: encodeURI(decodeURI(link)),
```
Comment (nschonni) [inline lib/proto/http.js:31]: The encoding passes should be opt-in so that people can catch places were they forgot to encode their URLs rather than silently passing because of this

Comment (NicolasMassart) [reply lib/proto/http.js:31]: It's not for people forgetting to encode but to support unicode.
Url supports unicode but on the server side, escaped unicode is required.
See #23

File: package-lock.json (hunk 1 above)
Comment (NicolasMassart) [reply package-lock.json:4]: I can revert but it seems compatible isn't it?

**Review by ThunderSon — APPROVED** [review]
Fixes the issue, and is the least absurd solution. Didn't find any better as well.
